### PR TITLE
Integration testing

### PR DESCRIPTION
**Pull request by ullrichd21**
# Integration testing

## Description

Adds a fixture for changing the working directory to a specific test assignment and adds a test for checking the output of the full system.

## Type of Change

- [x] Feature
- [ ] Bug fix
- [ ] Documentation

## Contributors

- @ullrichd21

## Blocked by

- #73
- #65
- #68

## Reminder

 - All GitHub Actions should be in a passing state before any pull request is merged.
 - All PRs must be reviewed by at least one team member and one member of the Integration team!
 - Any issues this PR closes are tagged in the description!

### CODE REVIEW

File: tests/test_main.py
Comment (burgess01) [inline tests/test_main.py:56]: Could you add a docstring and comments to this test?

File: tests/conftest.py
Comment (burgess01) [inline tests/conftest.py:6]: Could you add a docstring and comments to this to make this fixture more followable?

File: tests/conftest.py
Comment (burgess01) [inline tests/conftest.py:1]: Does this file need a docstring?

**Review by burgess01 — CHANGES_REQUESTED** [review]
This looks great! I just have a few things I think it would be helpful for you to add before this is merged!

**Review by Michionlion — APPROVED** [review]
This looks great to me! It will probably have to wait to be merged until we have the full functionality merged into `main`.

**Review by burgess01 — DISMISSED** [review]

**Comment by ullrichd21** [comment]
Do we want to parametrize this more or should we call it good with just this one for the initial release? 

@burgess01 @watson02 @laurennevill @mariakimheinert @noorbuchi

**Review by Michionlion — APPROVED** [review]
This should be good to go, pending one other review!

File: tests/test_assignment/gatorgrade.yml
Comment (mariakimheinert) [inline tests/test_assignment/gatorgrade.yml:28]: We should wait until we have support for shell checks to test them in our integration tests.

File: tests/test_main.py
Comment (mariakimheinert) [inline tests/test_main.py:21]: Are these debugging statements? If so, and they are needed, could we add comments to clarify why they are necessary?

File: tests/test_main.py
Comment (mariakimheinert) [inline tests/test_main.py:52]: These strings seem a bit brittle--for example, we may want to change the symbol for a passing check or the number of `=` characters. It might be better to check only for behaviors we really care about maintaining, such as that the descriptions listed in the `gatorgrade.yml` appear.

File: tests/test_main.py
Comment (mariakimheinert) [inline tests/test_main.py:44]: Since the second parameter is a list of strings we want in our output, might it be better to name the second parameter something like `expected_outputs`?

**Review by Michionlion — APPROVED** [review]

**Review by mariakimheinert — APPROVED** [review]

**Pull request merged**
